crates/ruff/src/rules/pyupgrade/rules/use_pep604_annotation.rs

```diff
@@ -75,6 +75,32 @@ enum TypingMember {
     Optional,
 }
 
+pub fn use_pep604_annotation_for_isinstance_or_issubclass(
```

Comment:
Sorry to go back on my own suggestion, but on second thought, let's make this its own rule(`use_pep604_isinstance`). It is part of PEP 604, but it's not related to _annotations_.